iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/tap/RatioTapPositionModification.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.modification.tap;
+
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.RatioTapChanger;
+import com.powsybl.iidm.network.RatioTapChangerHolder;
+
+/**
+ * @author Nicolas PIERRE <nicolas.pierre at artelys.com>
+ */
+public class RatioTapPositionModification extends AbstractTapPositionModification {
+
+    /**
+     * @param transfoId   the ID of the two windings transformer, which holds the rtc
+     * @param tapPosition the new tap position
+     */
+    public static RatioTapPositionModification createTwoWindingsRtcPosition(String transfoId, int tapPosition) {
+        return new RatioTapPositionModification(transfoId,
+            TransformerType.TWO_WINDINGS_TRANSFORMER, tapPosition, null);
+    }
+
+    /**
+     * @param transfoId   the ID of the three windings transformer, which holds the rtc
+     * @param tapPosition the new tap position
+     * @param legNumber   Defines the leg on which to apply the change for three winding tranformers. 0, 1 or 2
+     */
+    public static RatioTapPositionModification createThreeWindingsRtcPosition(String transfoId, int tapPosition,
+                                                                              int legNumber) {
+        return new RatioTapPositionModification(transfoId,
+            TransformerType.THREE_WINDINGS_TRANSFORMER, tapPosition, legNumber);
+    }
+
+    /**
+     * @param transfoId   the ID of the three windings transformer, which holds the rtc
+     * @param tapPosition the new tap position
+     * @param leg         defines on which leg of the three winding transformer the modification will be done. null on two windings.
+     */
+    public RatioTapPositionModification(String transfoId, TransformerType element,
```

Comment:
We cannot choose this design. When a user gives the id of a transformer, you cannot ask him to give the type of transformer : two or three windings. As you need the network to know, most of the checks you are doing must be in the apply. And for me you must have a method where we don't provide the leg number, without saying that it is a two windings transfromer.

Reply:
And finally the leg number must be optional. Even for a three windings transformer, you need it when you found more than one RTC in the transformer.